osu.Game/Overlays/Mods/ModColumn.cs

```diff
@@ -0,0 +1,409 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Humanizer;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Colour;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Utils;
+using osuTK;
+using osuTK.Graphics;
+using osuTK.Input;
+
+#nullable enable
+
+namespace osu.Game.Overlays.Mods
+{
+    public class ModColumn : CompositeDrawable
+    {
+        private Func<Mod, bool>? filter;
+
+        public Func<Mod, bool>? Filter
```

Comment:
In general we use the `public void Filter(FilterCriteria criteria)` structure (in at least [two](https://github.com/ppy/osu/blob/9532454e2a0d691652f40b7e15c3bd3503d9c9b4/osu.Game/Overlays/Music/Playlist.cs#L28) [places](https://github.com/ppy/osu/blob/eb25730b614ececc5f6672e298dc5e786984c92e/osu.Game/Screens/Select/BeatmapCarousel.cs#L548)) although arguably there's no interface governing this.

The same could be done here for conformity I think. Personally I find providing a criteria class feels more robust than a function (and reads a bit better), but if you have reasoning for writing it this way I'm also fine with it. Just needs xmldoc explaining it.

Reply:
It's written this way to link up with [the existing stuff](https://github.com/ppy/osu/blob/c8b975639834369e6e34c0ee6b3812761cfccaf7/osu.Game/Overlays/Mods/ModSelectOverlay.cs#L63) at a later date. In particular the free mod selection overlay leverages a free form delegate in a way that may be difficult to define in a "criteria" class.

Reply:
Aha, that makes sense. I guess sprinkle some xmldoc to clarify usage and it should be good.

Reply:
xmldoc has been added, please check if you find it legible.